charts/topolvm/values.yaml

```diff
@@ -0,0 +1,184 @@
+
+  # OPTIONAL: tune the node scoring
+  # Ref: https://github.com/topolvm/topolvm/blob/master/deploy/README.md
+  schedulerOptions: {}
+#    default-divisor: 1
+#    divisors:
+#      ssd: 1
+#      hdd: 10
+  options:
+    listen:
+      host: localhost
+      port: 9251
+
+
+
+# CSI node service
+node:
+  # Container Security Context
+  # ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  securityContext:
+    privileged: true
+
+  metrics:
+    enabled: true
+    annotations:
+      prometheus.io/port: "8080"
+
+
+# CSI controller service
+controller:
+  replicaCount: 2
+  ## terminationGracePeriodSeconds
+  ##
+  terminationGracePeriodSeconds: 10
+  ## Define resources requests and limits for single Pods.
+  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
+  resources:
+    requests:
+      memory: "50Mi"
+      cpu: "50m"
+    limits:
+      memory: "200Mi"
+      cpu: "200m"
+
+  # The update strategy to apply to the Deployment
+  ##
+  updateStrategy:
+    rollingUpdate:
+      maxSurge: 50%
+      maxUnavailable: 50%
+
+  # minReadySeconds
+  ##
+  minReadySeconds: 0
+
+  ## Affinity and anti-affinity
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  ##
+  affinity:
+    podAntiAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+            matchExpressions:
+              - key: app.kubernetes.io/name
+                operator: In
+                values:
+                  - controller
+          topologyKey: "kubernetes.io/hostname"
+
+  ## Node tolerations for server scheduling to nodes with taints
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+  ##
+  tolerations: []
+
+  ## Node labels for controller pod assignment
+  ## Ref: https://kubernetes.io/docs/user-guide/node-selection/
+  ##
+  nodeSelector: {}
+
+
+
+# Whether to create storageclass
+# ref: https://kubernetes.io/docs/concepts/storage/storage-classes/
+storageclass:
+  create: true
+  name: topolvm-provisioner
+  # Default storage class for dynamic volume provisioning
+  # ref: https://kubernetes.io/docs/concepts/storage/dynamic-provisioning
+  defaultclass: true
+  # Additional annotations
+  annotations: {}
+  # Supported filesystems are: ext4, xfs, and btrfs.
+  fstype: "xfs"
+  # enables CSI drivers to expand volumes. This feature is available for Kubernetes 1.16 and later releases.
+  allowVolumeExpansion: true
+  # volumeBindingMode can be either WaitForFirstConsumer or Immediate. WaitForFirstConsumer is recommended because TopoLVM cannot schedule pods wisely if volumeBindingMode is Immediate.
+  volumeBindingMode: WaitForFirstConsumer
+  additionalParameters: {}
+#    "topolvm.cybozu.com/device-class": "ssd"
+
+
+
+# Container Security Context
+# ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+securityContext:
+  runAsUser:  10000
+  runAsGroup: 10000
```

Comment:
Maybe these values can be used directly in the psp to tighten it up

Reply:
I think leaving the user to decide will be more generic
If someone wants to run as root, so he could, no ?

Reply:
yeah. I mean, use {{ .Values.securityContext.runAsUser }} and group in the psp to tighten up the uid / groups allowed to run as. Then whatever the user specifies here is applied to the psp.